### ISSUE TITLE
Just Stupid Question

### ISSUE THREAD

**Issue by saifulapm** [AndrewRadev/Vimfiles#3]
Hello @AndrewRadev, Just curiosity, Why you don't use Neovim instead of vim ? In Neovim 0.5, has many lua plugins which not supported by vim. I am new in vim world that's little awkward question ask. 

Anyway, Thanks for your awesome vim plugins.

**Comment by AndrewRadev** [comment]
> Anyway, Thanks for your awesome vim plugins.

Thank you for your kind words :).

Moving to neovim is not that easy for me. It doesn't support `+textprop` so this won't work: https://github.com/AndrewRadev/Vimfiles/blob/350f93b2bb8bf2c14747762d8c47e8a8bf23a613/ftplugin/ruby/private_methods.vim. It has a different popup API that gives me trouble: https://github.com/AndrewRadev/quickpeek.vim/issues/3#issuecomment-611113471. There's likely going to be more things that I don't know about that I'd run into that might be fixed in the future, but who knows.

I could migrate if there were big benefits, but to be honest, I don't see any, at least for myself. A big new neovim plugin is telescope, which is a fuzzy finder, but I don't really use fuzzy finders, I prefer more precise navigation like "find me a controller that starts with `user`" or something like that. Most of the other plugins I've seen are rewrites of existing ones in lua that are supposedly faster, but I don't really see the need for that speed-up. I have 121 plugins installed in my `bundle` directory:

``` .sh-session
% ls bundle | wc -l
121
```

My Vim boots in 175ms, which is fast enough for me:
```
175.037  000.001: --- VIM STARTED ---
```

If I was on an older computer, maybe I'd need speedups, but I suppose I'd start by cleaning up my plugins, because I'm confident I don't use all of them 😅. (Like, say, the "nyancat" plugin)

I do have some plugins that can be slow sometimes, but I don't imagine lua would help, since 1) it would require a full rewrite, and 2) I think a lot of them are slow for different reasons. For example, my own "sideways" plugin can be slow on large lists, but I don't think that's because of the language, but because it needs to jump character by character, inspect syntax groups, etc. Even in lua, the same calls to Vim would have to be made. I might be wrong, but rewriting the whole thing in lua is not something I'd feel like doing just to check :D. On the other hand, performance on fuzzy-finding is something that could be improved with a faster language, since I'm guessing there's a lot of logic that needs to be run very often, but I don't know if I have plugins that have that problem.

(I will say that native fuzzy matching has recently been added to Vim: [`:help matchfuzzy()`](https://vimhelp.org/eval.txt.html#matchfuzzy%28%29). But I can't really judge how well it could help with telescope-like plugins -- as I said, I don't really use fuzzy matching myself.)

Neovim has tree-sitter support, which could maybe be used for refactoring or something, but I've mostly seen syntax support from it, and I'm happy with Vim's standard highlighting. It might be more interesting in the future, but it's a tool, not a feature -- it depends on what people build with it. The built-in LSP client is something I'm not interested in, because... It's kind of hard to explain. LSP is something I have mixed feelings about, it's very powerful, but also somehow too inconsistently useful. For example, rust-analyzer takes 30+ seconds to boot on a project of mine, and if I want to check documentation or navigate the project, I really don't want to wait for it (which is why I started [this hacky thing](https://github.com/AndrewRadev/rustbucket.vim)). I feel like sitting down and writing something that interfaces with rust-analyzer, but gives me fine-grained control, maybe falls back to collecting information via regexes or ctags... I don't know, it's just a collection of vague ideas.

In short, migrating is something that would take a lot of work and I don't, personally, see any benefit from it, and there would be lots of drawbacks. I know that other people like some of the new plugins and/or enjoy writing lua, and that's great, I just don't happen to be one of them :). I'm sure that LSP, telescope and such are great tools for new users that can just turn them on and get used to them, but I already have a set of workflows that are not very compatible with them. Depending on what happens in the future, I might reconsider, but at this point, I've got a lot of "legacy" in my config and the cost of migrating seems pretty high.
